libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -586,18 +614,59 @@ export class FunctionUtils {
      * @param maxArgsLength The max length of a given function.
      */
     public static determineLocale(args: unknown[], maxArgsLength: number, locale = 'en-us'): string {
-        if (args.length === maxArgsLength && typeof args[maxArgsLength - 1] === 'string') {
-            locale = args[maxArgsLength - 1] as string;
+        if (args.length === maxArgsLength) {
+            const lastArg = args[maxArgsLength - 1];
+            if (typeof lastArg === 'string') {
+                locale = lastArg;
+            }
         }
 
         return locale;
     }
 
+        /**
+     *
+     * @param args An array of arguments.
+     * @param format A format string.
+     * @param locale A locale string.
+     * @param maxArgsLength The max length of a given function.
+     */
+    public static determineFormatAndLocale(
+        args: unknown[],
+        maxArgsLength: number,
+        format: string,
+        locale = 'en-us'
+    ): { format: string; locale: string } {
+        if (maxArgsLength >= 2) {
+            if (args.length === maxArgsLength) {
+                const lastArg = args[maxArgsLength - 1];
+                const secondLastArg = args[maxArgsLength - 2];
+                if (typeof lastArg === 'string' && typeof secondLastArg === 'string') {
+                    format =
+                        secondLastArg !== ''
+                            ? FunctionUtils.timestampFormatter(secondLastArg)
+                            : FunctionUtils.DefaultDateTimeFormat;
+                    locale = lastArg.substr(0, 2); //dayjs only support two-letter locale representattion
+                }
+            } else if (args.length === maxArgsLength - 1) {
+                const lastArg = args[maxArgsLength - 2];
+                if (typeof lastArg === 'string') {
+                    format = FunctionUtils.timestampFormatter(lastArg);
+                }
+            }
+        }
+
+        return { format: format, locale: locale };
+    }
+
     /**
      * Timestamp formatter, convert C# datetime to day.js format.
      * @param formatter C# datetime format
      */
     public static timestampFormatter(formatter: string): string {
+        if (formatter === undefined || formatter === '') {
```

Comment:
May as well do `if (!formatter)` as no falsy value is useful, right?

Reply:
changed